File: packages/core/src/event-type.js
let eventTypeCounter = 0;
/**
 * Create a named EventType declaration.
 * EventType is a channel — not an instance.
 */
export function createEventType(name) {
    return {
        name: `${name}#${eventTypeCounter++}`,
        _consumers: new Set(),
    };
}
/** Reset the counter (for testing) */
export function resetEventTypeCounter() {
    eventTypeCounter = 0;
}

File: packages/dart/lib/src/event_type.dart
import 'rule.dart';

int _eventTypeCounter = 0;

/// A named event channel declaration. Not an instance.
///
/// [T] is the payload type carried by events on this channel.
class EventType<T> {
  final String name;
  final Set<Rule> consumers = {};

  EventType._(this.name);

  @override
  String toString() => 'EventType($name)';
}

/// Create a named [EventType] declaration.
EventType<T> createEventType<T>(String name) {
  return EventType<T>._('$name#${_eventTypeCounter++}');
}

/// Reset the counter (for testing).
void resetEventTypeCounter() {
  _eventTypeCounter = 0;
}
